protocol/contracts/templegold/TempleGoldVesting.sol

```diff
@@ -0,0 +1,381 @@
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        if (_schedule.amount == 0) { return 0; }
+        if (block.timestamp <= _schedule.start) { return 0; }
+        if (_schedule.revoked) { return 0; }
+        uint32 _elapsed = _getElapsedTime(_schedule.start, uint32(block.timestamp), _schedule.duration);
+        return TempleMath.mulDivRound(_schedule.amount, _elapsed, _schedule.duration, false);
+    }
+
+    /**
+     * @notice Get vesting summary
+     * @param _from From timestamp
+     * @param _to To timestamp
+     * @return summary Vesting summary 
+     */
+    function getVestingSummary(uint32 _from, uint32 _to) external view override returns (VestingSummary[] memory summary) {
+        bytes32[] memory ids = _vestingIds.values();
+        uint256 _length = ids.length;
+        summary = new VestingSummary[](_length);
+        VestingSchedule memory _schedule;
+        for (uint256 i; i < _length; ++i) {
+            _schedule = schedules[ids[i]];
+            if (_schedule.start >= _from && _schedule.start + _schedule.duration <= _to) {
+                summary[i] = VestingSummary(
+                    _schedule.recipient,
+                    _schedule.distributed,
+                    uint128(_calculateTotalVestedAt(_schedule, uint32(block.timestamp))),
+                    uint128(_calculateTotalVestedAt(_schedule, _to))
+                );
+            }
+        }
+    }
+
+    /**
+     * @notice Get vesting Ids
+     * @return ids Vesting Ids 
+     */
+    function getVestingIds() external view override returns (bytes32[] memory ids) {
+        ids = _vestingIds.values();
+    }
+
+    /**
+     * @notice Get recipient vesting count
+     * @param _recipient Recipient address
+     * @return Count
+     */
+    function getRecipientVestingCount(address _recipient) external view override returns (uint256) {
+        return _holdersVestingCount[_recipient];
+    }
+
+    /**
+     * @notice Get releasable amount
+     * @param _vestingId Vesting Id
+     * @return Amount
+     */
+    function getReleasableAmount(bytes32 _vestingId) external view override returns (uint256) {
+        if(!vestingIdExists(_vestingId)) { return 0; }
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        return _calculateReleasableAmount(_schedule);
+    }
+
+    /**
+     * @notice Get total amount vested for an Id at timestamp
+     * @param _vestingId Vesting Id
+     * @param _at At timestamp
+     * @return Total vested
+     */
+    function getTotalVestedAt(bytes32 _vestingId, uint32 _at) external view override returns (uint256) {
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        if (_schedule.amount == 0) { return 0; }
+        if (block.timestamp <= _schedule.start) { return 0; }
+        return _calculateTotalVestedAt(_schedule, _at);
+    }
+
+    function _calculateReleasableAmount(
```

Comment:
I don't think you need this `memory` version - the two places this is being called should be passing in `storage` too